desktop/core/src/desktop/lib/rest/http_client.py

```diff
@@ -199,7 +200,11 @@ def execute(self, http_method, path, params=None, data=None, headers=None, allow
         self.logger.warn("GET and DELETE methods do not pass any data. Path '%s'" % path)
         data = None
 
+    if headers and 'timeout' in headers:
+      timeout = int(headers['timeout'])
+
```

Comment:
PS: should we move the LOG.debug here instead? 

'Overriding timeout xxx via header value xx'?

Reply:
Fixed